### ISSUE TITLE
Crashes if trying to copy search result with `copy-pipe` to pbcopy if result is on first terminal line, without scrollback

### ISSUE THREAD

**Issue by bcgraham** [tmux/tmux#2228]
`uname -sp`: Darwin i386
`tmux -V`: tmux next-3.2
`echo $TERM`: screen-256color
I can provide logs if you need, they have no additional information.

Recreated on e16191dbfc8efa0ce51356911f03ad8d88d09e28 (current master at this writing) and e71c5efd58b5160864ce851a4cbc93e5fde93685.

reproduce:
1. `env XDG_CONFIG_HOME=/tmp/tmux HOME=/tmp/tmux tmux -v -L foo bind -T copy-mode-vi y send -X copy-pipe 'pbcopy' \; new -s bar`
2. `search-backward` for something on top and first line of terminal, like a string in your shell prompt.

tmux crashes, `server exited unexpectedly`.

When I briefly looked into it, I was crashing in `window-copy.c`, line 3152 ([link](https://github.com/tmux/tmux/blob/master/window-copy.c#L3151-L3152)). `at` was something small and `sx` was large, `py` became zero, that's when the debug statements stopped printing so I'm assuming `at % (0 * sx)` was the culprit.

**Comment by bcgraham** [comment]
I can reproduce on a particular local setup, but only sometimes (for example, in iTerm2 with fish, but not with Terminal and zsh). I'll reopen when/if I can make it more reproducible.

**Comment by nicm** [comment]
This may be fixed now by 4589297e43ad4b385c9110879aab8338d7c45474. Thanks!

**Comment by bcgraham** [comment]
Yep, that fixed it. Thanks a lot!

For posterity, this was how to reproduce: 
```
XDG_CONFIG_HOME=/tmp/some_temp_dir HOME=/tmp/some_temp_dir tmux new 'echo foo; sh' \; run 'sleep 0.1' \; copy-mode \; send-keys -X search-backward foo \; send-keys -X copy-selection
```
